samcli/lib/build/app_builder.py

```diff
@@ -441,8 +441,13 @@ def _build_layer(
                     # Only set to this value if specified workflow is makefile
                     # which will result in config language as provided
                     build_runtime = compatible_runtimes[0]
+                image = None
+                if layer_name in self._build_images:
+                    image = self._build_images.get(layer_name)
+                elif None in self._build_images:
```

Comment:
What does `None` represent in `self._build_images`? This seems like an odd line to me. Is `None` some signal meaning the default image to use? Is so, we would change this to improve the readability. It is super unclear why we would do a look up in a dictionary with a key of `None`.

Reply:
This was discussed in the previous PR, #2725, None key represents the global image for all functions/layers. How do you suggest to change it? Involving @weichw into this discussion too

Reply:
this makes sense, but I would suggest to add a comment here explaining this because `None` representing the global image is quite implicit

Reply:
the entire code block (including the `image = None`) can be simplified as

```py
image = self._build_images.get(function_name, self._build_images.get(None))
```

Reply:
I would still argue for the original logic, a get inside a get doesn't seem as straight forward

Reply:
I have an alternative for you then:
```py
global_image = self._build_images.get(None)
image = self._build_images.get(layer_name, global_image)
```

Reply:
updated

Reply:
cool, thx

Reply:
> This was discussed in the previous PR, #2725, None key represents the global image for all functions/layers. How do you suggest to change it? Involving @CoshUS into this discussion too

I don't really know what has been discussed in previous prs. I am just looking at what is in this review and what we are adding/changing.

My suggestion is to make this more readable through being explicit. Getting `None` from a dictionary is very strange to me, especially because I assumed this is a `str:str` dictionary instead of mixing types. Someone should be able to understand the code by looking/read the function/class. Jumping between files to see where things were set makes it very difficult to follow code. In this case, I assumed we were getting the build image that corresponds to a specific layer or function. `None` in this case, seems "random" and doesn't indicate what/why we are getting this key. 

Ways to solve this and improve the readability:
1. pass a default image to use if one is not within `self._build_images`
2. Create `self._build_images` in a way that is consistent. So every function/layer is a key in `self._build_image` and whoever is creating this dictionary is responsible for setting it up that way.
3. Abstract this away into some class or function that hides the detail. So instead of interacting with the dictionary, we can create an indirection through a method that anyone can call. This moves the `None` logic into one place and provides a much more readable solution.

If you stay with the accessing of `None` (which I recommend we don't), you should be adding a comment that clearly explains why we are accessing `None` in a `build_images` dictionary.

Reply:
We have a default value for build image, the None key's value is a customer override of the default one, while the function/layer specific key value is the override of the None key one. This structure will not change. Adding a comment to explain